File: 278-first-bad-version/278-first-bad-version.c
// The API isBadVersion is defined for you.
// bool isBadVersion(int version);

int firstBadVersion(int n) {
    unsigned int l,r,mid;
    r = n;
    l = 1;
    
    while(l <= r)
    {
        mid = (l+r)/2;
        if(isBadVersion(mid) == true)
            r = mid - 1;
        else
            l = mid + 1;
    }
    return l;
}

File: 278-first-bad-version/278-first-bad-version.java
/* The isBadVersion API is defined in the parent class VersionControl.
      boolean isBadVersion(int version); */

public class Solution extends VersionControl {
    public int firstBadVersion(int n) {
        long l,r,mid;
        l = 1;
        r = n;
        
        while(l <= r){
            mid = (l+r)/2;
            if(isBadVersion((int)mid) == true)
                r = mid-1;
            else
                l = mid+1;
        }
        return (int)l;
    }
}